src/streams.ts

```diff
@@ -42,3 +43,36 @@ export function compose(streams: NodeJS.ReadWriteStream[]) {
     return new PassThrough({objectMode: true});
   }
 }
+
+/**
+ * A stream that passes through Vinyl file objects and loads any files
+ * that have not yet had their content loaded.
+ */
+export class VinylHydrateStream extends Transform {
+
+  constructor() {
+    super({ objectMode: true });
+  }
+
+  _transform(
+    file: File,
+    encoding: string,
+    callback: (error?: Error, data?: File) => void
+  ): void {
+    // Don't load a file that already has contents.
+    if (file.contents) {
+      callback(null, file);
+      return;
+    }
+
+    fs.readFile(file.path, (err?: Error, data?: Buffer) => {
```

Comment:
As discussed in person, we rely on this stream to coordinate asynchronous file loading that doesn't block or interfere with the analyzer stream.